R/init.R

```diff
@@ -222,27 +186,246 @@ init <- function(data,
     )
   }
 
-  ns <- NS(id)
-  # Note: UI must be a function to support bookmarking.
-  res <- list(
-    ui = function(request) {
-      ui_teal(
-        id = ns("teal"),
-        modules = modules,
-        title = title,
-        header = header,
-        footer = footer
+  res <- list()
+  res$landing_popup_server <- NULL
+
+  if (!is.null(title)) {
+    checkmate::assert_multi_class(title, c("shiny.tag", "shiny.tag.list", "html", "character"))
+    lifecycle::deprecate_warn(
+      when = "0.15.3",
+      what = "init(title)",
+      details = "Use `modify_title()` on the teal app object instead."
+    )
+  }
+  if (!is.null(header)) {
+    checkmate::assert_multi_class(header, c("shiny.tag", "shiny.tag.list", "html", "character"))
+    lifecycle::deprecate_warn(
+      when = "0.15.3",
+      what = "init(header)",
+      details = paste(
+        "Use `modify_header()` on the teal app object instead."
       )
-    },
-    server = function(input, output, session) {
-      if (!is.null(landing_popup)) {
-        do.call(landing_popup$server, c(list(id = "landing_module_shiny_id"), landing_popup$server_args))
-      }
-      srv_teal(id = ns("teal"), data = data, modules = modules, filter = deep_copy_filter(filter))
+    )
+  }
+  if (!is.null(footer)) {
+    checkmate::assert_multi_class(footer, c("shiny.tag", "shiny.tag.list", "html", "character"))
+    lifecycle::deprecate_warn(
+      when = "0.15.3",
+      what = "init(footer)",
+      details = paste(
+        "Use `modify_footer()` on the teal app object instead."
+      )
+    )
+  }
+
+  # argument transformations
+  ## `modules` - landing module
+  landing <- extract_module(modules, "teal_module_landing")
+  if (length(landing) == 1L) {
+    res$landing_popup_server <- landing[[1L]]$server
```

Comment:
We can use `add_custom_server` here and move this `if` after `server <- `
```suggestion
    app <- add_custom_server(app, landing[[1L]]$server)
```

Reply:
I think the if has to stay above because it removes the landing popup from the `modules`.

Reply:
Let me rearrange and make use of `add_custom_server`